### DISCUSSION TITLE
SmartCacheDatset does not see the whole train data when cache_rate is less than 1

### DISCUSSION THREAD

**Discussion by YazdanSalimi** [Project-MONAI/MONAI#6820] [Q&A] [answered]
Thank you for the great platform you developed.
I am reproducing the spleen 3d segmentation example from monai tutorials. 
The only thing I change is the train dataset from CacheDataset to SmartChachDataset.
the original code is : train_ds = CacheDataset(data=train_files, transform=train_transforms, cache_rate=1.0, num_workers=4)
The modified code is: train_ds = SmartCacheDataset(data=train_files, transform=train_transforms, cache_rate=.5, num_workers=0)

after this change, if the chach_rate is less than one (e.g., 0.1) the training loop only sees .1 of the cases. this problem does not exist with CacheDataset. Because this is the exact code of one of the tutorials, I did not copy my code here.
Thank you

**Comment by KumoLiu** [comment, marked as answer]
Hi @YazdanGIT, thanks for your interest here.
That, I believe, is the purpose of this dataset.
https://github.com/Project-MONAI/MONAI/blob/6f5cea3795278c20568261ebb38907d94b576b98/monai/data/dataset.py#L936-L957

(Convert to discussion for now, feel free to create another one if you still have some issues.)

**Reply by YazdanSalimi** [reply]
Thank you for the response.
What I observed by SmartChachDataset was loading [omage1, image2, image3, image4] for every epoch. I tested it by printing the image["image_meta_dict"]["file_name_or_obj"] every time and it only sees the images the same as the 1st epoch.
Thank you.
